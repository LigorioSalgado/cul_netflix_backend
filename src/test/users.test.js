const {graphql} =  require('graphql')

const {schema} =  require('../index');

const Users =  require('../schemas/Users');

const setupTest = require('./helpers');

const mutationRegister = `

    mutation Register($first_name:String!,$last_name:String!,$email:String!,
        $password:String!){
        signup(data:{first_name:$first_name,
            last_name:$last_name,
            email:$email,
            password:$password
        }){
            token
        }
    }
`

const queryMe = `

    query Profile{
        me{
            first_name,
            last_name,
            email
        }
    }

`

describe("Register user works correctly", () => {

    beforeEach( async () => await setupTest())


    it("Should create user correctly",async () => {
        const first_name = "Prueba"
        const last_name = "Prueba"
        const email = "prueba@email.com"
        const password = " miprueba"

        const res = await graphql(schema,mutationRegister,null,{}
                                ,{first_name,last_name,email,password})
    
        //expect(res).toMatchSnapshot();
        expect(res).toHaveProperty("data")
        expect(res.data).toHaveProperty("signup")
        expect(res.data.signup).toHaveProperty("token")


    })


    it("Should retrieve me profile",async () => {

        const me =  await Users.create({first_name:"prueba",last_name:"prueba",email:"prueba@prueba.com",password:"prueba"})

        const res  =  await graphql(schema,queryMe,null,{user:me},{})
        
        expect(res).toMatchSnapshot()
        expect(res.data.me.first_name).toBe(me.first_name)

    })

} )