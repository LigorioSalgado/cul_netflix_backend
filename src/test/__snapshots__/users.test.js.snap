// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register user works correctly Should retrieve me profile 1`] = `
Object {
  "data": Object {
    "me": Object {
      "email": "prueba@prueba.com",
      "first_name": "prueba",
      "last_name": "prueba",
    },
  },
}
`;
